cabal-install/Distribution/Client/ProjectOrchestration.hs

```diff
@@ -466,16 +472,135 @@ printPlan verbosity
+      [ (pkg, reason)
+      | (pkgid, reason) <- failures
+      , case reason of
+          DependentFailed {} -> False
+          _                  -> True
+      , InstallPlan.Configured pkg <-
+           maybeToList (InstallPlan.lookup plan pkgid)
+      ]
+
+    -- Special case: we don't want to report anything complicated in the case
+    -- of just doing build on the current package, since it's clear from
+    -- context which package failed.
+    --
+    -- We generalise this rule as follows:
+    --  - if only one failure occurs, and it is in a single root package (ie a
+    --    package with nothing else depending on it)
+    --  - and that failure is of a kind that always reports enough detail
+    --    itself (e.g. ghc reporting errors on stdout)
+    --  - then we do not report additional error detail or context.
+    --
+    isSimpleCase
+      | [(pkgid, reason)] <- failures
+      , [pkg]             <- rootpkgs
+      , installedUnitId pkg == pkgid
+      , isFailureSelfExplanatory reason
+      = True
+      | otherwise
+      = False
+
+    isFailureSelfExplanatory (BuildFailed e)
+      | Just (ExitFailure _) <- fromException e = True
+
+    isFailureSelfExplanatory (ConfigureFailed e)
+      | Just (ExitFailure _) <- fromException e = True
+
+    isFailureSelfExplanatory _                  = False
+
+    rootpkgs =
+      [ pkg
+      | InstallPlan.Configured pkg <- InstallPlan.toList plan
+      , hasNoDependents pkg ]
+
+    ultimateDeps pkgid =
+        filter (\pkg -> hasNoDependents pkg && installedUnitId pkg /= pkgid)
+               (InstallPlan.reverseDependencyClosure plan [pkgid])
+
+    hasNoDependents :: HasUnitId pkg => pkg -> Bool
+    hasNoDependents = null . InstallPlan.revDirectDeps plan . installedUnitId
+
+    renderFailure pkg reason =
+        case reason of
+          DownloadFailed  e -> "failed to download " ++ pkgstr ++ "."
+                            ++ showException e
+          UnpackFailed    e -> "failed to unpack " ++ pkgstr ++ "."
+                            ++ showException e
+          ConfigureFailed e -> "failed to build " ++ pkgstr ++ ". The failure"
+                            ++ "occurred during the configure step."
+                            ++ showException e
+          BuildFailed     e -> "failed to build " ++ pkgstr ++ "."
+                            ++ showException e
+          TestsFailed     e -> "tests failed for " ++ pkgstr ++ "."
+                            ++ showException e
+          InstallFailed   e -> "failed to build " ++ pkgstr ++ ". The failure"
+                            ++ "occurred during the final install step."
+                            ++ showException e
+
+          -- This will never happen, but we include it for completeness
+          DependentFailed pkgid -> " depends on " ++ display pkgid
+                                ++ " which failed to install."
+          PlanningFailed -> " failed during the planning phase."
```

Comment:
So, I grepped for PlanningFailed. Seems like we don't use it at all? (There's a similarly named constructor `cabal-install/Distribution/Client/BuildReports/Anonymous.hs` but I think they are unrelated.)